openedx/features/lti_course_tab/tab.py

```diff
@@ -34,7 +34,38 @@ class LtiCourseLaunchMixin:
     }
     DEFAULT_ROLE = 'Student'
 
+    def _get_pii_lti_parameters(self, course: CourseBlock, request: HttpRequest) -> Dict[str, str]:
+        """
+        Get LTI parameters that contain PII.
+
+        Args:
+            course (CourseBlock): CourseBlock object.
+            request (HttpRequest): Request object for view in which LTI will be embedded.
+
+        Returns:
+            Dictionary with LTI parameters containing PII.
+        """
+        lti_config = self._get_lti_config(course)
+        ask_to_send_username = lti_config.lti_config.get('ask_to_send_username', False)
+        ask_to_send_email = lti_config.lti_config.get('ask_to_send_email', False)
+        pii_config = {}
+        if ask_to_send_username:
+            pii_config['person_sourcedid'] = request.user.username
```

Comment:
Are these two keys, `person_sourcedid` and `person_contact_email_primary` correct?  I see something a little different here:

https://www.imsglobal.org/spec/lti/v1p3/migr#from-post-parameters-to-claims

The LTI v1.1 parameters are prefixed with `lis_` in that documentation.

[curious] Also this is LTI v1.1 specific, the claims are different in v1.3... should we account for that in the code here, even if we're not yet implementing it?

Reply:
@davidjoy I don't have deep familiarity with the spec, but am following the approach used by the LTI XBlock as seen [here](https://github.com/edx/xblock-lti-consumer/blob/8f7c40f5a8a6465a3211692bf43978af5dc5fc38/lti_consumer/lti_xblock.py#L986-L997).

I tried to break down the different aspects of building the LTI parameters so that we could potentially use a different mixin for LTI1.3 that provides similar parameters under different names. I think there are other more significant differences in their launches that might be a bigger issue first.

Reply:
Do we know at this point (in code) what version of the config is in use (like `lti_config.version`)?
We could at least conditionalize so that it only sets the params if we're on v1.1...

Reply:
Sure, that can be done.